kani/kani.py

```diff
@@ -454,4 +454,4 @@ def _auto_truncate_message(self, msg: ChatMessage, max_len: int) -> ChatMessage:
             "Auto truncate could not find an appropriate place to chunk the text. The returned value will be the first"
             f" {max_len} characters."
         )
-        return ChatMessage(role=msg.role, name=msg.name, content=full_content[: max_len - 3] + "...")
+        return msg.copy_with(text=full_text[: max_len - 3] + "...")
```

Comment:
This breaks if we have other message parts that contribute length (since `copy_with` will truncate the `text` but will preserve the `parts` thus len parts + text can still be > max -- I think?). If not, then auto truncate is erasing the parts here, which I also think would be unintended behavior. Either way, here's a rough fix:
```
max_text_len = max_len
if msg.parts:
    parts_len = self.message_token_len(msg.copy_with(text=""))
    max_text_len -= parts_len
return msg.copy_with(text=full_text[: max_text_len - 3] + "...")
```
Feel free to change this up as necessary.

Reply:
Ok on further inspection I realize that the intended behavior of `copy_with` seems to be to set `content` equal to either text or parts depending on what is passed in. i.e. `msg.copy_with(text=msg.text)` would not equal the original message if the message had parts.

That means that auto-truncate condenses all parts into text regardless of where its truncating. Is this accurate? Regardless, we should double check this code.